docs: Custom resource icon (#4041)
coder/coder: docs/templates/resource-metadata.md

}
```

## Using custom resource icon

> You can use custom icons for your resources using the `icon` attribute. The `icon` attribute needs to be a valid path or URL.

Reviewer: Some wording suggestions, hoping to keep the style of other sections

```suggestion
To use custom icons on your resources, use the `icon` attribute (must be a valid path or URL):
```

```hcl
resource "coder_metadata" "resource_with_icon" {
  count = data.coder_workspace.me.start_count
  resource_id = kubernetes_service_account.user_data.id
  icon = "/icon/database.svg"
  item {
    key = "name"
    value = kubernetes_deployment.coder[0].metadata[0].name
  }
}
```

> To make easier for you to customize your resource we added some built-in icons:

Reviewer: Love it! ❤️
